lib/model/CEventRatePopulationModel.cc

```diff
@@ -459,12 +461,26 @@ void CEventRatePopulationModel::sample(core_t::TTime startTime,
 
             for (auto& attribute : attributeValuesAndWeights) {
                 std::size_t cid = attribute.first;
+
+                const auto createAndAddAnnotation =
+                    [&](core_t::TTime t, const std::string& annotation) {
+                        m_CurrentBucketStats.s_Annotations.emplace_back(
+                            t, annotation, gatherer.searchKey().detectorIndex(),
+                            gatherer.searchKey().partitionFieldName(),
+                            gatherer.partitionFieldValue(),
+                            gatherer.searchKey().overFieldName(),
+                            gatherer.attributeName(cid),
+                            gatherer.searchKey().byFieldName(), "");
```

Comment:
You could use `EMPTY_STRING` instead of `""`.  It's a static constant in the base class and avoids a call to `strlen` on the `const char*`.